File: C/exercicio5.c
#include <stdio.h>
#include <string.h>

void inverter_string(char string[]) {
    int tamanho = strlen(string);
    for (int i = tamanho - 1; i >= 0; i--) {
        printf("%c", string[i]);
    }
}

int main() {
    char entrada[100];
    printf("Digite uma string: ");
    scanf("%s", entrada);
    printf("String invertida: ");
    inverter_string(entrada);
    return 0;
}

File: java/exercicio5.java
import java.util.Scanner;

// Inverter String
public class exercicio5 {
    public static void inverterString(String str) {
        char[] caracteres = str.toCharArray();
        for (int i = caracteres.length - 1; i >= 0; i--) {
            System.out.print(caracteres[i]);
        }
    }

    // Exemplo de uso
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Digite uma string: ");
        String entrada = scanner.nextLine();
        System.out.print("String invertida: ");
        inverterString(entrada);
        scanner.close();
    }
}
